### PR TITLE
Add stf load/store tests to the regress target

### PR DESCRIPTION
**Pull request by StanSofomoDev**
PR for issue https://sofomo.atlassian.net/browse/CN-46

1. Run `regress` target to rebuild Dromajo and run CTests
2. Run `update_test_files` target to regenerate .efl files used in tests + filesystem (after this step, golden files will be outdated and need to be updated)

### CODE REVIEW

**Comment by StanSofomoDev** [comment]
@jeffnye-gh I fixed the issue with linux tests failing, I believe that PR is ready to be reviewed.

File: tests/stf_load_store/src/main.c [hunk 1]
```diff
@@ -37,76 +54,84 @@
...
-        //  lwu rd, offset(rs1)
-        "lb t0, 0(sp);"           // Load byte from stack into t0 (signed)
-        "lbu t0, 0(sp);"          // Load byte from stack into t0 (unsigned)
-        "lh t0, 0(sp);"           // Load halfword from stack into t0 (signed)
-        "lhu t0, 0(sp);"          // Load halfword from stack into t0 (unsigned)
-        "lw t0, 0(sp);"           // Load word from stack into t0 (signed)
-        "lwu t0, 0(sp);"          // Load word from stack into t0 (unsigned)
-        "ld t0, 0(sp);"           // Load doubleword from stack into t0
+        //FIXME: there should be one of each of these: 
+
```
Comment (jeffnye-gh) [inline tests/stf_load_store/src/main.c:85]: Can you go ahead and remove the invalid cases? 

Looking at it now I'd rather not keep the obviously incorrect ones around

File: tests/stf_load_store/src/main.c [hunk 2]
```diff
@@ -37,76 +54,84 @@
...
-        //  sd rs2, offset(rs1)
-        //  sh rs2, offset(rs1)
-        //  sw rs2, offset(rs1)
-        "sb t0, 0(sp);"           // Store byte from t0 to stack
-        "sh t0, 0(sp);"           // Store halfword from t0 to stack
-        "sw t0, 0(sp);"           // Store word from t0 to stack
-        "sd t0, 0(sp);"           // Store doubleword from t0 to stack
+        // FIXME: should be one of each of these:
+
+        //"c.sb t0, 0(sp);"    //x c.sb rs2, offset(rs1)
```
Comment (jeffnye-gh) [inline tests/stf_load_store/src/main.c:117]: similarly

**Review by jeffnye-gh — APPROVED** [review]
please make the requested changes and merge, 

the changes are simple, we won't need another round of review

File: tests/stf_load_store/CMakeLists.txt [hunk 3]
```diff
@@ -2,33 +2,67 @@ cmake_minimum_required(VERSION 3.10)
 
 project(stf_load_store)
 
-# Gather source files
+set(COMPILER_BARE "/data/tools/riscv64-unknown-elf/bin/riscv64-unknown-elf-gcc")
+set(COMPILER_LINUX "/data/tools/riscv64-unknown-linux-gnu/bin/riscv64-unknown-linux-gnu-gcc")
+
```
[the remark this answers is not in the document]
Comment (StanSofomoDev) [reply tests/stf_load_store/CMakeLists.txt:7]: Baremetal/Linux build commands have it's respective compilers added in CMake as a dependency, so when it's not found CMake will complain about it:

image

File: tests/stf_load_store/src/main.c (hunk 1 above)
Comment (StanSofomoDev) [reply tests/stf_load_store/src/main.c]: Done

File: tests/stf_load_store/src/main.c (hunk 2 above)
Comment (StanSofomoDev) [reply tests/stf_load_store/src/main.c]: Done

File: tests/stf_load_store/common/crt.S [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # See LICENSE for license details.
 
-#include "encoding.h"
+#define MSTATUS_MPP         0x00001800
```
[the remark this answers is not in the document]
Comment (StanSofomoDev) [reply tests/stf_load_store/common/crt.S:3]: Done.

**Pull request merged**